src/analytics/formErrorTracker/formErrorTracker.ts

```diff
@@ -24,69 +24,95 @@ export class FormErrorTracker extends FormTracker {
     }
 
     const form = document.forms[0];
-    if (!form) {
+    let fields: FormField[] = [];
+
+    if (form && form.elements) {
+      fields = this.getErrorFields(form);
+    } else {
       return false;
     }
 
-    if (!form.elements) {
+    if (!fields.length) {
+      console.log("form or fields not found");
       return false;
     }
-
-    const submitUrl = this.getSubmitUrl(form);
-    const formErrorTrackerEvent: FormEventInterface = {
-      event: this.eventType,
-      event_data: {
-        event_name: this.eventName,
-        type: validateParameter(this.getType(form), 100),
-        url: validateParameter(submitUrl, 100),
-        text: validateParameter(this.getErrorMessage(), 100),
-        section: validateParameter(this.getFieldLabel(), 100),
-        action: "error",
-        external: "undefined",
-        link_domain: this.getDomain(submitUrl),
-        "link_path_parts.1": this.getDomainPath(submitUrl, 0),
-        "link_path_parts.2": this.getDomainPath(submitUrl, 1),
-        "link_path_parts.3": this.getDomainPath(submitUrl, 2),
-        "link_path_parts.4": this.getDomainPath(submitUrl, 3),
-        "link_path_parts.5": this.getDomainPath(submitUrl, 4),
-      },
-    };
-
     try {
-      this.pushToDataLayer(formErrorTrackerEvent);
+      for (const field of fields) {
+        const submitUrl = this.getSubmitUrl(form);
```

Comment:
submitUrl has always the same value. You don't need to set it for every fields.
Only once is enough.

Reply:
done